lib/ansible_tower_client/hash_model.rb

```diff
@@ -83,7 +85,8 @@ def add_accessor_methods(method, key)
 
     def initialize(json_or_hash)
       raw_hash = json_or_hash.kind_of?(Hash) ? json_or_hash : JSON.parse(json_or_hash)
-      @data = Hash[raw_hash.collect { |key, value| [key, self.class.send(:convert_value, key, value, self)] }]
+      raw_hash = raw_hash.collect { |key, value| [key, self.class.send(:convert_value, key, value, self)] }
+      @data = ActiveSupport::HashWithIndifferentAccess[raw_hash]
```

Comment:
And is it needed?

Reply:
Here is the explanation:
```
$ bin/console 

2.3.0 :001 > jt = AnsibleTowerClient::JobTemplate.new(nil, :ask_limit_on_launch => true)
 => <AnsibleTowerClient::JobTemplate ask_limit_on_launch=true> 
2.3.0 :002 > jt.ask_limit_on_launch
 => true 

2.3.0 :003 > jt = AnsibleTowerClient::JobTemplate.new(nil, 'ask_limit_on_launch' => true)
 => <AnsibleTowerClient::JobTemplate ask_limit_on_launch=true> 
2.3.0 :004 > jt.ask_limit_on_launch
 => nil 
```
However, I solved that in other way in PR update.

Reply:
`JobTemplate` object is always created from the JSON response of the API request. They key is always a string not symbol. 

My test is the opposite
```
irb(main):011:0> jt = AnsibleTowerClient::JobTemplate.new(nil, 'ask_limit_on_launch' => true)
=> <AnsibleTowerClient::JobTemplate ask_limit_on_launch=true>
irb(main):012:0> jt.ask_limit_on_launch
=> true
irb(main):013:0> jt = AnsibleTowerClient::JobTemplate.new(nil, :ask_limit_on_launch => true)
=> <AnsibleTowerClient::JobTemplate ask_limit_on_launch=true>
irb(main):014:0> jt.ask_limit_on_launch
=> nil
```

Reply:
I did that for reason. Not just because I wanted to.
I have no idea, what happens with your test.
My tests are failing without this change. 
Even the specs are failing under a certain conditions.
If you wish to see what's happening - let's do Bluejeans.

Reply:
How are you going to handle this trait? https://github.com/ansible/ansible_tower_client_ruby/blob/cf77e3bfa94222d3ddb1c59cd6eda332ec31f92b/spec/factories/responses.rb#L47
It makes that attribute a symbol.

Reply:
I now understand the cause of the above observations. In reality the constructor is not a hash but the JSON string from API response. The keys are strings not symbols. There are not mix uses of strings and symbols.

Reply:
And your test shows the same.

Reply:
And the main problem with specs:
When you run my test alone, it will work:
`$ bundle exec rspec spec/job_template_spec.rb:31`
https://github.com/ansible/ansible_tower_client_ruby/blob/cf77e3bfa94222d3ddb1c59cd6eda332ec31f92b/spec/job_template_spec.rb#L31
But if you run tests all together, it will fail, because of that string vs symbol mess.

If you'll comment that line:
https://github.com/ansible/ansible_tower_client_ruby/blob/cf77e3bfa94222d3ddb1c59cd6eda332ec31f92b/spec/job_template_spec.rb#L11
Or simply "xit" this line:
https://github.com/ansible/ansible_tower_client_ruby/blob/cf77e3bfa94222d3ddb1c59cd6eda332ec31f92b/spec/support/shared_examples/job_template.rb#L2
Test will pass.

Reply:
In reply to your: https://github.com/ansible/ansible_tower_client_ruby/pull/119#discussion_r211289385
My change only transforms all the keys to strings, and also transforms attribute names to strings too (to handle `jt.some_property`). So your algo will keep work as you wanted.